File: app/src/main/java/io/mgba/Presenter/Interfaces/IGamesPresenter.java
package io.mgba.Presenter.Interfaces;


import io.mgba.Data.Database.Game;
import io.mgba.UI.Activities.Interfaces.ILibrary;
import io.reactivex.functions.Consumer;

public interface IGamesPresenter {

    void onDestroy();

    int getPlatform();

    void loadGames(ILibrary databaseHelper);
    void onRefresh(ILibrary iLibrary);

    Consumer<Game> getOnClick();

}

File: app/src/main/java/io/mgba/presenter/interfaces/IGamesPresenter.kt
package io.mgba.presenter.interfaces


import io.mgba.data.database.model.Game
import io.mgba.ui.activities.interfaces.ILibrary

interface IGamesPresenter {

    val platform: Int

    fun onClick(game: Game): Any
    fun onDestroy()

    fun loadGames(databaseHelper: ILibrary)
    fun onRefresh(iLibrary: ILibrary)

}
